api/v1alpha1/etcdcluster_types.go

```diff
@@ -30,6 +30,8 @@ type EtcdClusterSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
+	// etcd configuration options are passed as environmental vars to the etcd container, refer to etcd documentation for configuration options application for the version of etcd being used.
+	EtcdOptions []EtcdOption `json:"etcdOptions,omitempty"`
```

Comment:
Aren't these options "optional"? Wonder it `// +optional` Kubebuilder marker makes sense here

Reply:
I thought that using `omitempty` would make it optional, but I'll test with the `// +optional` marker

Reply:
I did some testing here, using `omitempty` does make the field optional. With the current configuration the CRD has EtcdOptions as an optional parameter. 

Using `omitempty` to make fields optional does appear to be the convention for this project so I've left this as is